### PR TITLE
Add foundation

### PR DESCRIPTION
**Pull request by wavtype**
## Description

added foundation to /wavtype/foundation

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wavtype/foundation/keymaps/via/via.json [hunk 1]
```diff
@@ -0,0 +1,279 @@
+{
```
Comment (zvecr) [inline keyboards/wavtype/foundation/keymaps/via/via.json:1]: This file should be deleted as via config files live in another repo.

File: keyboards/wavtype/foundation/readme.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+make wavtype/foundation:via:flash
+
+See the build environment setup and the make instructions for more information. Brand new to QMK? Start with our Complete Newbs Guide.
+
+Bootloader
+Enter the bootloader in 3 ways:
+
+Bootmagic reset: Hold down the key at (0,1) in the matrix (Escaoe key) and plug in the keyboard
+Physical reset button: Briefly press the button on the back of the PCB
+Keycode in layout: Press the key mapped to RESET if it is available (by default this is set to "Fn + R")
```
Comment (zvecr) [inline keyboards/wavtype/foundation/readme.md:20]:
```suggestion
# foundation

The foundation pcb was designed for protozoa studio's foundation line of keyboards.

* Keyboard Maintainer: wavtype

Make example for this keyboard (after setting up your build environment):

    make wavtype/foundation:default

Flashing example for this keyboard:

    make wavtype/foundation:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:
* **Bootmagic reset**: Hold down the key at (0,1) in the matrix (Escaoe key) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB
* **Keycode in layout**: Press the key mapped to RESET if it is available (by default this is set to "Fn + R")
```

File: keyboards/wavtype/foundation/config.h [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x03a7
+#define PRODUCT_ID   0x0003
+#define DEVICE_VER   0x0001
+#define MANUFACTURER protozoa.studio
+#define PRODUCT      protozoa.studio foundation
+#define DESCRIPTION  protozoa.studio foundation
```
Comment (zvecr) [inline keyboards/wavtype/foundation/config.h:14]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/wavtype/foundation/keymaps/via/via.json (hunk 1 above)
Comment (wavtype) [reply keyboards/wavtype/foundation/keymaps/via/via.json:1]: thanks, missed that.

Comment (wavtype) [reply keyboards/wavtype/foundation/keymaps/via/via.json]: do i need to make a new pull request after the changes are made?

File: keyboards/wavtype/foundation/rules.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/wavtype/foundation/rules.mk:20]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

File: keyboards/wavtype/foundation/config.h [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x03a7
+#define PRODUCT_ID   0x0003
+#define DEVICE_VER   0x0001
+#define MANUFACTURER protozoa.studio
+#define PRODUCT      protozoa.studio foundation
```
Comment (fauxpark) [inline keyboards/wavtype/foundation/config.h:13]:
```suggestion
#define PRODUCT      foundation
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/wavtype/foundation/config.h [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+/*== or choose animations ==*/
+//#    define RGBLIGHT_EFFECT_BREATHING
+//#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//#    define RGBLIGHT_EFFECT_SNAKE
+//#    define RGBLIGHT_EFFECT_KNIGHT
+//#    define RGBLIGHT_EFFECT_CHRISTMAS
+//#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//#    define RGBLIGHT_EFFECT_RGB_TEST
+//#    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/wavtype/foundation/config.h:70]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/wavtype/foundation/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+	KC_F4,   KC_LSHIFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSHIFT,                    KC_UP,
+	KC_F5,   KC_LCTRL, KC_LGUI, KC_LALT,                KC_SPC,                KC_RALT, KC_RGUI, MO(1), KC_RCTRL,    KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
+    [1] = LAYOUT(
+    RGB_TOG,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,   KC_INS, KC_HOME, KC_PGUP,
+    RGB_MOD,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,              KC_TRNS, KC_TRNS, KC_TRNS,
+	RGB_HUI,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+	RGB_SAI,   KC_TRNS,      KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                  KC_VOLU,
+	RGB_VAI,   KC_TRNS, KC_TRNS, KC_TRNS,                KC_TRNS,                KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                    KC_MPRV, KC_VOLD, KC_MNXT
+    )
```
Comment (fauxpark) [inline keyboards/wavtype/foundation/keymaps/via/keymap.c:27]: Need two more empty layers here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**